nwplus/portal
# Portal schema changes

## Description
<details>
<summary>made `copyText` and `headerText` dynamic (the application dashboard page is fully dynamic now)</summary>

- removed safewalk option for HC

| Status | Image |
|--------|-------|
| inProgress | ![image](https://github.com/user-attachments/assets/55cdb75a-dc8f-4aa6-953e-4939a15e7cd5) |
| applied | ![image](https://github.com/user-attachments/assets/2bf641b7-16c1-4319-b7ae-140f4235c8c8) |
| waitlisted | ![image](https://github.com/user-attachments/assets/3f642c71-03bb-4697-bba2-b7da675542d2) |
| rejected | ![image](https://github.com/user-attachments/assets/c9ff81f4-7a1c-4cff-8bff-0df4167f9236) |
| acceptedNoResponseYet | ![image](https://github.com/user-attachments/assets/8c02c690-77ab-4e24-b078-fab9828d6ae7) |
| acceptedAndAttending | ![image](https://github.com/user-attachments/assets/95636789-2f09-4743-bd4a-9caad875b30e) |

</details>

<details>

<summary>updated livesite schema to:</summary>

```
{
   "hackathonStart":{
      "cmd-f":"2025-03-08T17:00:00.000Z",
      "nwhacks":"2025-01-18T17:00:00.000Z",
      "hackcamp":"2024-11-09T17:00:00.000Z"
   },
   "applicationsOpen":{
      "cmd-f":true,
      "nwhacks":false,
      "hackcamp":false
   },
   "hackathonEnd":{
      "cmd-f":"2025-03-10T01:00:00.000Z",
      "hackcamp":"2024-11-11T01:00:00.000Z",
      "nwhacks":"2025-01-20T01:00:00.000Z"
   },
   "notionLinks":{
      "nwhacks":{
         "preHackathonWorkshops":""
      },
      "cmd-f":{
         "preHackathonWorkshops":"https://nwplus.notion.site/PUBLIC-cmd-f-2024-Pre-Hackathon-Workshops-b2429635e65a4e79957a63bfd3d8b27d?pvs=4"
      },
      "hackcamp":{
         "preHackathonWorkshops":""
      }
   },
   "hackingStart":{
      "hackcamp":"2024-08-09T20:00:00.000Z",
      "cmd-f":"2025-03-08T20:00:00.000Z",
      "nwhacks":"2025-01-18T20:00:00.000Z"
   },
   "judgingOpen":{
      "nwhacks":false,
      "hackcamp":false,
      "cmd-f":true
   },
   "applicationDeadline":{
      "cmd-f":"February 22nd at 11:59 PM (Pacific Time)",
      "hackcamp":"October 31st at 11:59 PM (Pacific Time)",
      "nwhacks":"December 4th at 11:59 PM (Pacific Time)"
   },
   "lastEdited":{
      "seconds":1720119063,
      "nanoseconds":910000000
   },
   "rsvpBy":{
      "hackcamp":"November 5th at 11:59 PM (Pacific Time)",
      "cmd-f":"March 2nd at 11:59 PM (Pacific Time)",
      "nwhacks":"January 15th at 11:59 PM (Pacific Time)"
   },
   "hackathonWeekend":{
      "nwhacks":"January 18th-19th",
      "hackcamp":"November 9th-10th",
      "cmd-f":"March 8th-9th"
   },
   "hackingEnd":{
      "nwhacks":"2025-01-19T20:00:00.000Z",
      "hackcamp":"2024-11-10T20:00:00.000Z",
      "cmd-f":"2025-03-09T20:00:00.000Z"
   },
   "lastEditedBy":"",
   "judgingReleased":{
      "cmd-f":false,
      "hackcamp":false,
      "nwhacks":false
   },
   "sendAcceptancesBy":{
      "nwhacks":"early January",
      "cmd-f":"early March",
      "hackcamp":"early November"
   },
   "offWaitlistNotify":{
      "nwhacks":"January 15th at 11:59 PM (Pacific Time)",
      "hackcamp":"November 5th at 11:59 PM (Pacific Time)",
      "cmd-f":"March 2nd at 11:59 PM (Pacific Time)"
   },
   "submissionsOpen":{
      "cmd-f":false,
      "nwhacks":false,
      "hackcamp":false
   },
   "waiversAndForms":{
      "cmd-f":{
         "media":"https://docs.google.com/document/d/175Rsgz5HXUpfPUqNbRpPrJTmqqxFfhxZJBXNFMDX0SU/edit?usp=sharing",
         "nwMentorship":"https://forms.gle/nG9vwzc5HG5X5jVB8",
         "releaseLiability":" https://docs.google.com/document/d/1BfDr_S2FNNl7Xfv0fncZLzUV2pq0O91z6k6xwiN4gPw/edit?usp=sharing",
         "covid":"https://docs.google.com/document/d/1OlcOiEPSjngFKFEndG6wBBOYm02JTpY3KldeJ9GCfIo/edit?usp=sharing"
      },
      "nwhacks":{
         "nwMentorship":"",
         "media":"",
         "releaseLiability":"",
         "covid":""
      },
      "hackcamp":{
         "media":"",
         "releaseLiability":"",
         "nwMentorship":"",
         "covid":""
      }
   }
}
```

</details>

- added `waiversAndForms` (moved out of `Constants.js`) and `notionLinks` to the schema since these will be different every hackathon

Note: leaving `rsvpOpen` alone for now ([slack thread](https://nwplus2024-2025.slack.com/archives/C071PT1LS4U/p1722204568475469))


## Manual testing
- toggle the different feature flags (all the ones that end in `Open`) and make sure that the app reflects the changes
